[brvndonkoh] iP

brvndonkoh opened the pull request:
# Clarawr Chatbot 🤖  

Clarawr is a simple chatbot that helps users **manage their tasks efficiently**. Users can add three types of tasks:  
- **ToDos**  
- **Deadlines**  
- **Events**  

Clarawr also allows users to **mark tasks as done** ✅ and keep track of their progress.  

## Features  
1. Add tasks (`todo`, `deadline`, `event`)  
2. Mark tasks as **done**  
3. View all tasks in a list  
4. Save and load tasks automatically  

## Usage Example  
Here’s an example of adding a **deadline** task in Clarawr:  
```java
Clarawr clarawr = new Clarawr();
clarawr.addTask("deadline submit report /by 2025-02-10");

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

brvndonkoh committed:
Add greeting

brvndonkoh committed:
Rename main class

brvndonkoh committed:
Add support for echo

brvndonkoh committed:
Added Add, List

brvndonkoh committed:
Added Add, List

brvndonkoh committed:
Added Task Class and mark/unmark feature

brvndonkoh committed:
Usage of inheritance for Level 4

brvndonkoh committed:
Usage of inheritance for Level 4

brvndonkoh committed:
Usage of inheritance for Level 4

brvndonkoh committed:
Handling Exceptions where user types in incomplete information

brvndonkoh committed:
Now we can delete/remove tasks with Clarawr :D

brvndonkoh committed:
Clarawr.java: Add save feature
Task.java: Add new method

brvndonkoh committed:
Add enum

brvndonkoh committed:
Clarawr.java: Add save feature

brvndonkoh committed:
Merge branch 'branch-Level-7'

brvndonkoh committed:
Add proper 24 hour clock

Deadline.java: Return date and time in MMM-DD-YYYY format
Event.java: Return date and time in MMM-DD-YYYY format

brvndonkoh committed:
Clarawr.java: Add new instruction

listByDate allows the user to search for tasks on the specified date

brvndonkoh committed:
Merge branch 'branch-Level-8'

brvndonkoh committed:
Remove debugging lines

brvndonkoh committed:
Deadline.java: Made local variable deadline final

TaskList.java: Made local variable tasks final

brvndonkoh committed:
Merge branch 'A-MoreOOP'

brvndonkoh committed:
Move all classes into Clarawr package

brvndonkoh committed:
Merge branch 'A-Packages'

brvndonkoh committed:
Merge branch 'add-gradle-support'

brvndonkoh committed:
Merge branch 'add-gradle-support' of https://github.com/brvndonkoh/ip into add-gradle-support

brvndonkoh committed:
Add Gradle

brvndonkoh committed:
Add TaskType

brvndonkoh committed:
Merge branch 'add-gradle-support'

brvndonkoh committed:
Add 2 JUnit test

brvndonkoh committed:
Merge branch 'A-JUnit-branch'

brvndonkoh committed:
Add JavaDoc

brvndonkoh committed:
Add find

brvndonkoh committed:
Follow the coding standard

brvndonkoh committed:
Merge branch 'branch-A-JavaDoc'

brvndonkoh committed:
Resolve merge conflicts for JavaDoc

brvndonkoh committed:
Merge branch 'branch-A-JavaDoc'

brvndonkoh committed:
Resolved merge conflicts between JavaDoc

brvndonkoh committed:
Merge branch 'master' of https://github.com/brvndonkoh/ip

brvndonkoh committed:
Resolevd merge conflicts with branch-Level-9

weiyanminoo commented on src/main/java/Clarawr/Clarawr.java:
```diff
@@ -0,0 +1,257 @@
+package Clarawr;
```
Your main class currently has alot of responsibilities so you can consider separate certain parts of your code to different classes (ill add comments at the respective sections but up to u haha i think this is fine too)

weiyanminoo commented on src/main/java/Clarawr/Clarawr.java:
```diff
@@ -0,0 +1,257 @@
+
+import java.util.Scanner;
+
+enum TaskType {
```
can consider separating this part into an enum

weiyanminoo commented on src/main/java/Clarawr/Clarawr.java:
```diff
@@ -0,0 +1,257 @@
+     *
+     * @param instruction The user input command.
+     * @throws ClarawrException If an error occurs while processing the command.
+     */
```
logic wise this is fine but for better readability you can consider placing this into the parser class

weiyanminoo commented on src/main/java/Clarawr/Clarawr.java:
```diff
@@ -0,0 +1,257 @@
+     *
+     * @param description The description of the Todo task.
+     * @throws ClarawrException If the description is empty.
+     */
```
i think it might be better if you add this and the subsequent methods into the TaskList class as they have the same responsibility of task management

weiyanminoo commented on src/main/java/Clarawr/Todo.java:
```diff
@@ -0,0 +1,40 @@
+     *
+     * @param description A brief description of the task.
+     * @param isDone A boolean indicating whether the task is completed.
+     */
```
the constructor directly modifies isDone which breaks the encapsulation from the Task class so you cna consider calling markAsDone() instead

weiyanminoo commented on src/main/java/Level_1/Clarawr.java:
```diff
@@ -0,0 +1,20 @@
+package Level_1;
+import java.util.Scanner;
+
```
i think this part is slightly confusing cos even though it is in a different package, you named this as Clarawr class too

weiyanminoo reviewed (commented):
hii brandon! overall good job on your code thus far! i just added some comments that could improve on the overall flow and readability but its not neccessary to change since your code logic is still correct. keep up the good work :))

Wailydest commented on src/main/java/Clarawr/Storage.java:
```diff
@@ -0,0 +1,59 @@
+ * The file path is hardcoded, and tasks are stored in a text file.
+ */
+public class Storage {
+    private static final String FILE_PATH = "C:\\Users\\user\\CS2103_IP\\ip\\src\\data\\clarawr.txt";
```
The file path is windows-specific, and it could fail when tried on MacOS. Maybe it would be better to use a system-independent way of achieving navigation?

Wailydest replied:
Also, it could be nice to have a way to pass a file path into Storage to make it more flexible instead of using hard-coded path. For example, when testing a Storage we'd want to make it save to a special testing directory instead of the default data one.

Wailydest commented on src/main/java/Clarawr/Clarawr.java:
```diff
@@ -0,0 +1,257 @@
+        String[] commandParts = parser.parseCommand(instruction);
+        String command = commandParts[0].toLowerCase();
+
+		switch (command) {
```
According to the coding [standards](https://se-education.org/guides/conventions/java/intermediate.html) given to us, the spacing within the switch() statement should be 4-spaces long. It seems that this code use tabs instead. Maybe you could consider changing your editor formatting settings?

Wailydest commented on src/test/java/Clarawr/TaskListTest.java:
```diff
@@ -0,0 +1,41 @@
+
+public class TaskListTest {
+    @Test
+    public void checkGetAllTask() {
```
Could tests be more comprehensive here? For example, the Storage class could be tested for it's ability to save and load from the storage

Wailydest commented on src/main/java/Clarawr/Storage.java:
```diff
@@ -0,0 +1,59 @@
+     *
+     * @return A list of tasks loaded from the file.
+     */
+    public ArrayList<Task> loadTasksFromFile() {
```
It's not quite clear what's the distinction between the member method `saveTasksToFile()` and the class method `loadTasksFromFile()`. Why one of them is static and the other is not, if the class don't have any member variables anyway? Maybe for consistency they both should have the same structure?

Wailydest reviewed (commented):
Good job! The structure of the project is quite transparent and it's easy to navigate over it. Few changes to it could still be made, including those specified by the previous reviewer. Also more testing units could definitely benefit the app. Overall, LGTM!

brvndonkoh committed:
Add GUI interface

brvndonkoh committed:
Merge branch 'branch-Level-10'

brvndonkoh committed:
Add core task management features and command handling

brvndonkoh committed:
Add assertions for null checks

brvndonkoh committed:
Refactor code to adhere to coding conventions

brvndonkoh committed:
Merge branch 'A-CodeQuality-branch'

brvndonkoh committed:
Merge conflicts between master and A-Assertions-Branch

brvndonkoh committed:
Add duplicate task detection to prevent adding duplicate task

brvndonkoh committed:
Merge branch 'C-DetectDuplicates-branch' to integrate duplicate
detection feature

- Incorporated the feature to check and prevent adding duplicate tasks
  in the task list.
- Ensured that tasks with identical descriptions are not added more thn
  once.

brvndonkoh committed:
CLarawr's responses do not fully match its baby lion personality

Some replies feel out of character and do not consistently reflect the playful and youthful nature of a baby lion. This affects immersion and user experience.

Adjust responses to specific inputs to better fit CLarawr's personality

Ensures consistency in tone and maintains a fun, engaging experience for users interacting with CLarawr.

Tested responses to ensure they align with the intended personality traits. No changes to underlying logic or structure.

brvndonkoh committed:
Merge branch 'A-Personality-branch'

brvndonkoh committed:
Add Clarawr icon and display name in top-right window

brvndonkoh committed:
Add new user and Clarawr icons to match the jungle vibe

brvndonkoh committed:
Add user guide to help new users familiarise themselves with the application

brvndonkoh committed:
Merge branch 'A-UserGuide-branch'

brvndonkoh committed:
Remove unused imports

brvndonkoh committed:
Remove unused imports

brvndonkoh committed:
Edit README.md

brvndonkoh committed:
Change package name to lowercase

brvndonkoh committed:
Change indent to follow coding convention

brvndonkoh committed:
Merge branch 'CodingConvetion-branch'

brvndonkoh committed:
Rename JAR file to lowercase

brvndonkoh committed:
Update README for more accurate instructions

brvndonkoh committed:
Add other OS in build.gradle

brvndonkoh committed:
Add trim() in handleCommand method

brvndonkoh committed:
Add some error handling

brvndonkoh committed:
Add more error handling
